grails-app/services/org/transmartproject/db/RestExportService.groovy

```diff
@@ -22,52 +23,23 @@ class RestExportService {
 
     ConceptsResource conceptsResourceService
 
+    int cohortNumberID
+
     List<File> export(arguments) {
         DataExportFetchTask task = dataExportFetchTaskFactory.createTask(arguments)
         task.getTsv()
     }
 
-    def formatDataTypes(List datatypes){
-        //Maybe make this a marshaller.
-        def returnDataTypeList = []
-        def dataTypesList = []
-        datatypes.each { dataTypeMap ->
-            dataTypeMap.each {key, value->
-                if (key in dataTypesList){
-                    returnDataTypeList.each { map ->
-                        if (key in map.values()){
-                            value.each{it.remove('datatypeCode')}
-                            map.get('cohorts').add([concepts:value])
-                        }
-                    }
-                } else{
-                    dataTypesList.add(key)
-                    def datatypeCode = value.collect({it ->
-                        it.get('datatypeCode')})
-                    value.each{it.remove('datatypeCode')}
-                    def datatypeMap = [dataType:key,
-                                       dataTypeCode: datatypeCode[0],
-                                       cohorts:[[concepts:value]]]
-                    returnDataTypeList.add(datatypeMap)
-                }
-            }
-        }
-        returnDataTypeList
-    }
-
-
-    def getDataTypes(List conceptKeysList){
-        List cohortDataTypes = []
-        Map datatypesMap = [:]
+    public def getDataTypes(List conceptKeysList, List dataTypes, Integer cohortNumber){
+        cohortNumberID = cohortNumber
         conceptKeysList.each { conceptKey ->
             OntologyTerm concept = conceptsResourceService.getByKey(conceptKey)
-            datatypesMap = getHighDimDataType(concept, datatypesMap)
+            dataTypes = getDataType(concept, dataTypes)
```

Comment:
was your intension here to create a collection of dataTypes and return this collection ?
If yes - then you could use .collect() {element-> getDatatype(...)} 
this forEach loop looks as if it was overwriting dataTypes over and over.